[Shipping Labels] Remove green checkmark icon from error

irfano opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->
WOOMOB-555

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
On shipment flow, error messages had ✅ icon. This PR removes the icon from only error messages but keep it for success messages.

### Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
#### Purchase error
1. To create fake endpoint response for error, add this to API faker
```
[
  {
    "request": {
      "id": 0,
      "path": "/wcshipping/v1/label/purchase/%",
      "queryParameters": [],
      "type": {
        "type": "wp-api"
      }
    },
    "response": {
      "body": "",
      "endpointId": 0,
      "statusCode": 500
    }
  }
]
```
2. Go to order and select and order.
3. Tap Create shipping labels button.
4. Fill the fields and tap Purchase label button.

#### Split shipment error
1. To create fake endpoint response for error, add this to API faker
```
[
  {
    "request": {
      "id": 0,
      "path": "/wcshipping/v1/shipments/%",
      "queryParameters": [],
      "type": {
        "type": "wp-api"
      }
    },
    "response": {
      "body": "",
      "endpointId": 0,
      "statusCode": 500
    }
  }
]
```
2. Go to order and select and order.
3. Tap Create shipping labels button.
4. Tap Split Shipment button.
5. Make some changes and tap DONE button.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
Steps above

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
|Before|After|
|-|-|
|<img src="https://github.com/user-attachments/assets/b25ef3b2-8599-457e-a749-0e2d73d8c09a" width=350>|<img src="https://github.com/user-attachments/assets/420a05a2-48a9-4ac7-abee-3bade9a44e80" width=350>|
|<img src="https://github.com/user-attachments/assets/0b582cc2-baa5-4176-a2e1-7271149006fd" width=350>|<img src="https://github.com/user-attachments/assets/d3a49243-bdab-4ae0-b49a-c74923a3a690" width=350>|

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

irfano committed:
Remove green icon error snackbars on shipment screen

irfano committed:
Remove green icon error snackbars on shipment creation screen

hichamboushaba reviewed (commented):
Thanks for working on this @irfano, while the changes work, I have a concern with the implementation and the roles.

Just to showcase the issue, apply the following [patch](https://github.com/user-attachments/files/20882717/Changes.patch), then open the Shipping Labels form, you'll notice that the `loading` snackbar is shown without the checkmark.

The above is a result of how the SnackbarHostState is supposed to work: it queues snackbar requests, but given we remove the `SnackbarHost` and replace it with a new one with each request, we might end up with wrong results.

This is not an issue with our current implementation, because with the usage of `LaunchedEffect`, each change of `snackbarData` will result in canceling the old request (which is a questionable behavior IMO that we should fix in the future, but it's a different topic), but I just wanted to showcase the issue, and how it might impact us if we change the behavior to respect the queue.

Normally, a `SnackbarHost` can show different types of Snackbars; we shouldn't need to replace it, we need only to decide on the `Snackbar` based on the `data` that it sends to its callback.

I did an attempt to solve this by creating a custom implementation of `SnackbarVisuals`, please check it out in this [patch](https://github.com/user-attachments/files/20882780/Use_SnackbarVisuals.patch) and let me know what you think.

irfano committed:
Use SnackbarVisuals for snackbars in create shipping label flow

irfano committed:
Use custom generic snackbar for shipping labels

Replaces the `SuccessSnackbar` with a more generic `ShippingLabelsSnackbar` to handle both standard and success snackbars. This allows for displaying an icon on the snackbar.

irfano commented:
Thank you for the great feedback. I’ve applied your patch.
Since we’re now implementing `SnackbarVisuals`, we can use its `hasIcon` parameter and show a single snackbar for both success and error cases. I’ve implemented this, could you please review it again?

hichamboushaba reviewed (approved):
> Since we’re now implementing SnackbarVisuals, we can use its hasIcon parameter and show a single snackbar for both success and error cases. I’ve implemented this, could you please review it again?

The `hasIcon` is a custom parameter, not part of the interface `SnackbarVisuals`, right? Just a confirmation.

But otherwise, this change is a great improvement, I would just suggest renaming `hasIcon` to `hasSuccessCheckmark` or something similar, as `hasIcon` is very generic, and could be confusing, while we have a single icon that we use.

irfano committed:
Rename `hasIcon` to `hasSuccessCheckmark` in Snackbar

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/261">22.7</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

irfano commented:
> The hasIcon is a custom parameter, not part of the interface SnackbarVisuals, right? Just a confirmation.

Yes, that’s right. I meant that we’re now able to pass a custom parameter via `ShippingLabelsSnackbarVisuals`, which implements `SnackbarVisuals`.
